services/src/messaging/src/main/java/org/openkilda/messaging/model/AbstractNetworkEndpoint.java

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.messaging.model;
+
+import org.openkilda.messaging.Utils;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+
+import java.io.Serializable;
+
+@Data
+public abstract class AbstractNetworkEndpoint implements Serializable {
+    @JsonProperty("switch-id")
+    private final String datapath;
+
+    @JsonProperty("port-id")
+    private final Integer portNumber;
+
+    @JsonCreator
+    public AbstractNetworkEndpoint(String datapath, Integer portNumber) {
+        this.datapath = datapath;
+        this.portNumber = portNumber;
+
+        validate();
```

Comment:
This is bad practice. "Constructor must not invoke overridable methods" (Item 17: Design and document for inheritance or else prohibit it. Effective Java, 2nd edition. J. Bloch)